apps/fxc-front/src/app/components/2d/path-element.ts

```diff
@@ -273,36 +249,27 @@ export class PathElement extends connect(store)(LitElement) {
       this.faiSectors = new FaiSectors();
       this.faiSectors.addListeners('rightclick', (e) => this.appendToPath(e.latLng));
     }
-    if (score.circuit == CircuitType.FlatTriangle || score.circuit == CircuitType.FaiTriangle) {
+    if (result.circuit == CircuitType.FlatTriangle || result.circuit == CircuitType.FaiTriangle) {
       this.faiSectors.update(result.turnpoints);
       this.faiSectors.setMap(this.map);
     } else {
       this.faiSectors.setMap(null);
     }
 
-    this.postScoreToHost(score);
-  }
-
-  private toScore(result: ScoringResult): Score {
-    return new Score({
-      circuit: result.circuit,
-      distanceM: result.lengthKm * 1000,
-      multiplier: result.multiplier,
-      closingRadiusKm: result.closingRadiusKm ?? 0,
-      indexes: result.solutionIndices,
-      points: result.score,
-    });
+    store.dispatch(plannerSlice.setScore(result));
+    // FFVL/CFD
+    this.postScoreToHost(result);
   }
 
   // Sends a message to the iframe host with the changes.
-  private postScoreToHost(score: Score) {
+  private postScoreToHost(scoringResult: ScoringResult) {
     let kms = '';
     let circuit = '';
-    if (score.distanceM && window.parent) {
-      kms = (score.distanceM / 1000).toFixed(1);
-      circuit = CIRCUIT_SHORT_NAME[score.circuit];
-      if (score.circuit == CircuitType.OpenDistance) {
-        circuit += score.indexes.length - 2;
+    if (scoringResult.lengthKm !== 0 && window.parent) {
+      kms = scoringResult.lengthKm.toFixed(1);
+      circuit = CIRCUIT_SHORT_NAME[scoringResult.circuit];
+      if (scoringResult.circuit === CircuitType.OpenDistance) {
+        circuit += scoringResult.solutionIndices.length - 2;
```

Comment:
Could you please make sure `score.indexes.length` and `scoringResult.solutionIndices.length` for all the open distance types for FFVL scoring?

Reply:
Sorry, I dont understand what you want. Is your sentence incomplete?

Nevertheless, may be it would be better to use `scoringResult.turnpoints.length` for the circuit name computation
```suggestion
        circuit += scoringResult.turnpoints.length;
```